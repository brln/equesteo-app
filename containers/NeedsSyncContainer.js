import React, { PureComponent } from 'react'
import { connect } from 'react-redux';

import {
  ActivityIndicator,
  StyleSheet,
  Text,
  View
} from 'react-native';

import Button from '../components/Button'
import { brand, lightGrey } from '../colors'
import BuildImage from '../components/Images/BuildImage'
import { doSync, startListeningFCM, switchRoot } from '../actions/functional'
import { FEED } from '../screens'


class NeedsSyncContainer extends PureComponent {
  static options() {
    return {
      topBar: {
        title: {
          text: "About",
          color: 'white',
          fontSize: 20
        },
        backButton: {
          color: 'white',
        },
        background: {
          color: brand,
        },
        elevation: 0
      }
    };
  }

  constructor (props) {
    super(props)
    this.state = {
      trying: false
    }
    this.doSync = this.doSync.bind(this)
  }

  doSync () {
    this.setState({
      trying: true
    })
    this.props.dispatch(doSync()).then(() => {
      if (this.props.syncFail) {
        this.setState({
          trying: false
        })
      } else {
        this.props.dispatch(switchRoot(FEED))
        this.props.dispatch(startListeningFCM())
      }
    })
  }

  render() {
    return (
      <View style={{height: '100%'}}>
        <View style={{
          flex: 1,
          backgroundColor: lightGrey,
          alignItems: 'center',
          justifyContent: 'center',
          flexDirection: 'column',
        }}>
          <BuildImage
            source={require('../img/logo250.png')}
            style={{
              width: 120,
              height: 120,
              alignItems: 'center',
            }}
          />
          <Text style={{
            fontFamily: 'Panama-Light',
            fontSize: 30,
            color: 'black',
          }}>
            Equesteo
          </Text>
        </View>
        <View style={styles.container}>
          <View style={{flex: 1}}>
            <Text style={{textAlign: 'center', fontSize: 25}}>Oh No!</Text>
            <Text style={{textAlign: 'center', fontSize: 14, paddingTop: 10}}>When you sign up, the app has to download some data, and we're having problems finishing that. Please try again when you have better service (or WiFi works best!).</Text>
            <Text style={{textAlign: 'center', fontSize: 14, paddingTop: 10}}>If you continue to have problems, please get in touch!</Text>
            <Text style={{textAlign: 'center', fontSize: 18, paddingTop: 15}}>info@equesteo.com</Text>
            <View style={{padding: 20}}>
              { this.state.trying ?
                <ActivityIndicator /> :
                <Button
                  onPress={this.doSync}
                  color={brand}
                  text={'Try Again'}
                />
              }
            </View>
          </View>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: lightGrey,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  errorBox: {
    textAlign: 'center',
    paddingTop: 5,
    paddingBottom: 5,
    backgroundColor: 'red',
  }
})

function mapStateToProps (state) {
  const localState = state.get('localState')
  return {
    error: localState.get('error'),
    syncFail: localState.get('fullSyncFail')
  }
}

export default connect(mapStateToProps)(NeedsSyncContainer)
