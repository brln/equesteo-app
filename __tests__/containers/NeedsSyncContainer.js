import configureStore from 'redux-mock-store'
import React from "react"
import {fromJS, Map} from "immutable"
import renderer from "react-test-renderer"
import { Provider } from 'react-redux'
import thunk from "redux-thunk"

import NeedsSyncContainer from '../../containers/NeedsSyncContainer'


const mockStore = configureStore([thunk])

test('render', () => {
  const store = mockStore(fromJS({
    pouchRecords: {
      users: {}
    },
    localState: {}
  }))

  const tree = renderer.create(
    <Provider store={store}>
      <NeedsSyncContainer />
    </Provider>)
  expect(tree.toJSON()).toMatchSnapshot()
})
