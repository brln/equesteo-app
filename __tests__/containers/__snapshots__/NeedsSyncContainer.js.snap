// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<View
  style={
    Object {
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e8eae8",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "alignItems": "center",
          "height": 120,
          "width": 120,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "Panama-Light",
          "fontSize": 30,
        }
      }
    >
      Equesteo
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e8eae8",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "padding": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 25,
            "textAlign": "center",
          }
        }
      >
        Oh No!
      </Text>
      <Text
        style={
          Object {
            "fontSize": 14,
            "paddingTop": 10,
            "textAlign": "center",
          }
        }
      >
        When you sign up, the app has to download some data, and we're having problems finishing that. Please try again when you have better service (or WiFi works best!).
      </Text>
      <Text
        style={
          Object {
            "fontSize": 14,
            "paddingTop": 10,
            "textAlign": "center",
          }
        }
      >
        If you continue to have problems, please get in touch!
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
            "paddingTop": 15,
            "textAlign": "center",
          }
        }
      >
        info@equesteo.com
      </Text>
      <View
        style={
          Object {
            "padding": 20,
          }
        }
      >
        <View
          accessible={true}
          clickable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#5bc0eb",
              "borderColor": undefined,
              "borderRadius": 10,
              "borderWidth": null,
              "marginTop": 10,
              "opacity": 1,
              "paddingBottom": 10,
              "paddingTop": 10,
              "width": undefined,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Try Again
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
